src/nvim/CMakeLists.txt

```diff
@@ -168,6 +170,8 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
   target_link_libraries(main_lib INTERFACE pthread c++abi)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
   target_link_libraries(nvim_bin PRIVATE -lsocket)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Haiku")
+  target_link_libraries(nvim_bin PRIVATE -lnetwork)
```

Comment:
probably cleaner to just add this to where the first target_link_libraries for haiku is on lin 154 (think you can even combine them into a single target_link_libraries if you use main_lib or nvim_bin for both of them)